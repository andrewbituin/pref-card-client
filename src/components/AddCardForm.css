.add-card-navigation{
    background-color:#51A4FB;
    color: white;
    padding: 20px;
}
.add-card-header{
    margin: 0px;
}
.add-card-title{
    margin: 6px;
    font-size: 3em;
}
.add-card-form{
    width: 75%;
    margin: 0 auto;
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    margin-bottom: 20px;
}

.add-card-section-one{
   margin: 20px 0px;
}
textarea{
    font-size: 14px;
    margin-bottom: 10px;
}
select{
    font-size: 16px;
}
.surgeon, .access, .gloveSize{
    margin: 5px 0 0 5px;
}

@media screen and (max-width: 600px){
    .add-card-form{
        width: 90%;
        margin: auto;
    }
}