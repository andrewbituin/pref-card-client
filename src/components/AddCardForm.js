import React from "react";
import { Link } from "react-router-dom";
import ApiService from "../services/api-service";
import CardsContext from "../context/CardsContext";
import LogoutButton from "./LogoutButton";
import "./AddCardForm.css";
import uuid from "uuid";

export default class AddCardForm extends React.Component {
  static contextType = CardsContext;

  state = {
    error: false
  };

  handleSubmit = e => {
    e.preventDefault();
    const obj = {};
    const form = new FormData(e.target);
    form.forEach((val, key) => (obj[key] = val));

    const user = this.context.usersList.find(
      user => user.full_name === obj.surgeon
    );
    obj.user_id = user.id;
    ApiService.postCard(JSON.stringify(obj))
      .then(this.context.addCard(obj))
      .then(() => this.props.history.push("/all"));
  };

  generateOptions = () => {
    const surgeons = this.context.usersList.filter(
      user => user.position === "doctor"
    );
    return surgeons.map(surgeon => {
      return <option key={surgeon.id}>{surgeon.full_name}</option>;
    });
  };
  generateAccessOptions = () => {
    const users = this.context.usersList.filter(
      user => user.position === "nurse"
    );
    return users.map(user => {
      return <option key={uuid()}>{user.full_name}</option>;
    });
  };
  generateForm = () => {
    return (
      <form className="add-card-form" onSubmit={e => this.handleSubmit(e)}>
        <section className="add-card-section-one">
          Surgeon:
          <select className="surgeon" name="surgeon">
            {this.generateOptions()}
          </select>
          <br />
          Edit Access:
          <select className="access" name="access">
            {this.generateAccessOptions()}
          </select>
          <br />
          Procedure:
          <br />
          <textarea rows="4" cols="50" className="procedure" name="procedure" />
          <br />
          Position:
          <br />
          <textarea rows="4" cols="50" className="position" name="position" />
          <br />
          Glove Size:
          <select className="gloveSize" name="glove_size">
            <option>6</option>
            <option>6.5</option>
            <option>7</option>
            <option>7.5</option>
            <option>8</option>
            <option>8.5</option>
          </select>
          <br />
          <label>
            Glove Type: <s />
            <label htmlFor="small">Small</label>
            <input
              type="radio"
              id="small"
              className="gloveType"
              value="small"
              name="glove_type"
              defaultChecked
            />
            <label htmlFor="medium">Medium</label>
            <input
              type="radio"
              id="medium"
              className="gloveType"
              name="glove_type"
              value="medium"
            />
            <label htmlFor="large">Large</label>
            <input
              type="radio"
              id="large"
              className="gloveType"
              name="glove_type"
              value="large"
            />
            <br />
          </label>
          <label>
            Dominant Hand: <s />
            <label htmlFor="right">Right</label>
            <input
              type="radio"
              id="right"
              className="dominantHand"
              name="dominant_hand"
              value="right"
              defaultChecked
            />
            <label htmlFor="left">Left</label>
            <input
              type="radio"
              id="left"
              className="dominantHand"
              name="dominant_hand"
              value="left"
            />
            <br />
          </label>
        </section>
        <section className="add-card-section-two">
          Equipment:
          <br />
          <textarea rows="4" cols="50" className="equipment" name="equipment" />
          <br />
          Supplies:
          <br />
          <textarea rows="4" cols="50" className="supplies" name="supplies" />
          <br />
          Instrumentation:
          <br />
          <textarea
            rows="4"
            cols="50"
            className="instrumentation"
            name="instrumentation"
          />
          <br />
          Suture and usage:
          <br />
          <textarea
            rows="4"
            cols="50"
            className="sutureAndUsage"
            name="suture_and_usage"
          />
          <br />
        </section>
        <section className="add-card-section-three">
          Dressings:
          <br />
          <textarea rows="4" cols="50" className="dressings" name="dressings" />
          <br />
          Skin Prep:
          <br />
          <textarea rows="4" cols="50" className="skinPrep" name="skin_prep" />
          <br />
          Medication:
          <br />
          <textarea
            rows="4"
            cols="50"
            className="medications"
            name="medications"
          />
          <br />
          <button type="submit">Submit</button>
        </section>
      </form>
    );
  };

  render() {
    return (
      <div className="add-card-container">
        <nav className="add-card-navigation">
          <header className="add-card-header">
            <h1 className="add-card-title">prefcards</h1>
            <Link to="/all">
              <button type="click" className="cancel-button">
                Cancel
              </button>
            </Link>
            <LogoutButton />
          </header>
        </nav>
        {this.generateForm()}
      </div>
    );
  }
}
