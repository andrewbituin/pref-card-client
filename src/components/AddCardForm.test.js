import React from "react";
import ReactDOM from "react-dom";
import AddCardForm from "./AddCardForm";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

enzyme.configure({ adapter: new Adapter() });

describe("AddCardForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <AddCardForm />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the UI as expected", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <AddCardForm />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    shallow(
      <MemoryRouter>
        <AddCardForm />
      </MemoryRouter>
    );
  });
});
