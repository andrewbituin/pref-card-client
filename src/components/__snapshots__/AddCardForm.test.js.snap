// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCardForm component renders the UI as expected 1`] = `
<div
  className="add-card-container"
>
  <nav
    className="add-card-navigation"
  >
    <header
      className="add-card-header"
    >
      <h1
        className="add-card-title"
      >
        prefcards
      </h1>
      <a
        href="/all"
        onClick={[Function]}
      >
        <button
          className="cancel-button"
          type="click"
        >
          Cancel
        </button>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          onClick={[Function]}
          type="click"
        >
          Logout
        </button>
      </a>
    </header>
  </nav>
  <form
    className="add-card-form"
    onSubmit={[Function]}
  >
    <section
      className="add-card-section-one"
    >
      Surgeon:
      <select
        className="surgeon"
        name="surgeon"
      />
      <br />
      Procedure:
      <br />
      <textarea
        className="procedure"
        cols="50"
        name="procedure"
        rows="4"
      />
      <br />
      Position:
      <br />
      <textarea
        className="position"
        cols="50"
        name="position"
        rows="4"
      />
      <br />
      Glove Size:
      <select
        className="gloveSize"
        name="glove_size"
      >
        <option>
          6
        </option>
        <option>
          6.5
        </option>
        <option>
          7
        </option>
        <option>
          7.5
        </option>
        <option>
          8
        </option>
        <option>
          8.5
        </option>
      </select>
      <br />
      <label>
        Glove Type: 
        <s />
        <label
          htmlFor="small"
        >
          Small
        </label>
        <input
          className="gloveType"
          defaultChecked={true}
          id="small"
          name="glove_type"
          type="radio"
          value="small"
        />
        <label
          htmlFor="medium"
        >
          Medium
        </label>
        <input
          className="gloveType"
          id="medium"
          name="glove_type"
          type="radio"
          value="medium"
        />
        <label
          htmlFor="large"
        >
          Large
        </label>
        <input
          className="gloveType"
          id="large"
          name="glove_type"
          type="radio"
          value="large"
        />
        <br />
      </label>
      <label>
        Dominant Hand: 
        <s />
        <label
          htmlFor="right"
        >
          Right
        </label>
        <input
          className="dominantHand"
          defaultChecked={true}
          id="right"
          name="dominant_hand"
          type="radio"
          value="right"
        />
        <label
          htmlFor="left"
        >
          Left
        </label>
        <input
          className="dominantHand"
          id="left"
          name="dominant_hand"
          type="radio"
          value="left"
        />
        <br />
      </label>
    </section>
    <section
      className="add-card-section-two"
    >
      Equipment:
      <br />
      <textarea
        className="equipment"
        cols="50"
        name="equipment"
        rows="4"
      />
      <br />
      Supplies:
      <br />
      <textarea
        className="supplies"
        cols="50"
        name="supplies"
        rows="4"
      />
      <br />
      Instrumentation:
      <br />
      <textarea
        className="instrumentation"
        cols="50"
        name="instrumentation"
        rows="4"
      />
      <br />
      Suture and usage:
      <br />
      <textarea
        className="sutureAndUsage"
        cols="50"
        name="suture_and_usage"
        rows="4"
      />
      <br />
    </section>
    Dressings:
    <br />
    <textarea
      className="dressings"
      cols="50"
      name="dressings"
      rows="4"
    />
    <br />
    Skin Prep:
    <br />
    <textarea
      className="skinPrep"
      cols="50"
      name="skin_prep"
      rows="4"
    />
    <br />
    Medication:
    <br />
    <textarea
      className="medications"
      cols="50"
      name="medications"
      rows="4"
    />
    <br />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
